tests/system/python/api/test_statistics.py

```diff
@@ -0,0 +1,184 @@
+    # Define the template file for fogbench
+    fogbench_template_path = os.path.join(
+        os.path.expandvars('${FOGLAMP_ROOT}'), 'data/{}'.format(TEMPLATE_NAME))
+    with open(fogbench_template_path, "w") as f:
+        f.write(
+            '[{"name": "%s", "sensor_values": '
+            '[{"name": "sensor", "type": "number", "min": %d, "max": %d, "precision": 0}]}]' % (
+                asset_name, SENSOR_VALUE, SENSOR_VALUE))
+
+    add_south(south_plugin, south_branch, foglamp_url, service_name=PLUGIN_NAME)
+
+    yield start_south_coap
+
+    # Cleanup code that runs after the caller test is over
+    remove_data_file(fogbench_template_path)
+    remove_directories("/tmp/foglamp-south-{}".format(south_plugin))
+
+
+class TestStatistics:
+
+    def test_cleanup(self, reset_and_start_foglamp):
+        # TODO: Remove this workaround
+        # Use better setup & teardown methods
+        pass
+
+    def test_default_statistics(self, foglamp_url):
+        conn = http.client.HTTPConnection(foglamp_url)
+        conn.request("GET", '/foglamp/statistics')
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        jdoc = json.loads(r)
+        assert len(jdoc), "No data found"
+        assert len(STATS_KEYS) == len(jdoc)
+        keys = [key['key'] for key in jdoc]
+        assert Counter(STATS_KEYS) == Counter(keys)
+
+    def test_default_statistics_history(self, foglamp_url):
+        conn = http.client.HTTPConnection(foglamp_url)
+        conn.request("GET", '/foglamp/statistics/history')
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        jdoc = json.loads(r)
+        assert len(jdoc), "No data found"
+        assert 15 == jdoc['interval']
+        assert {} == jdoc['statistics'][0]
+
+    def test_statistics_history_with_stats_collector_schedule(self, foglamp_url, wait_time, retries):
+        # wait for sometime for stats collector schedule to start
+        time.sleep(wait_time * retries)
+
+        conn = http.client.HTTPConnection(foglamp_url)
+        conn.request("GET", '/foglamp/statistics/history')
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        jdoc = json.loads(r)
+        assert len(jdoc), "No data found"
+        assert 15 == jdoc['interval']
+        assert 1 == len(jdoc['statistics'])
+        statistics = jdoc['statistics'][0]
+        assert Counter(STATS_HISTORY_KEYS) == Counter(statistics.keys())
+        del statistics['history_ts']
+        assert Counter([0, 0, 0, 0, 0, 0]) == Counter(statistics.values())
+
+    @pytest.mark.parametrize("request_params, keys", [
+        ('', STATS_HISTORY_KEYS),
+        ('?limit=1', STATS_HISTORY_KEYS),
+        ('?key=READINGS', {'history_ts', 'READINGS'}),
+        ('?key=READINGS&limit=1', {'history_ts', 'READINGS'}),
+        ('?key=READINGS&limit=0', {}),
```

Comment:
Add test for time based queries - minutes,hours,days

Reply:
https://github.com/foglamp/FogLAMP/pull/1483#discussion_r269464887